libevmasm/Assembly.cpp

```diff
@@ -705,18 +708,18 @@ LinkerObject const& Assembly::assemble() const
 		// Append an INVALID here to help tests find miscompilation.
 		ret.bytecode.push_back(uint8_t(Instruction::INVALID));
 
-	for (size_t i = 0; i < m_subs.size(); ++i)
+	for (auto const& sub: subRef)
 	{
-		auto references = subRef.equal_range(i);
-		if (references.first == references.second)
-			continue;
-		for (auto ref = references.first; ref != references.second; ++ref)
-		{
-			bytesRef r(ret.bytecode.data() + ref->second, bytesPerDataRef);
-			toBigEndian(ret.bytecode.size(), r);
-		}
-		ret.append(m_subs[i]->assemble());
+		vector<size_t> subIds = decodeSubIds(sub.first);
+		Assembly const* currentAssembly = this;
```

Comment:
Can you extract this into a helper?

Reply:
Or actually - why isn't this just part of `decodeSubIds`?

Reply:
Ok, maybe better to have its own function.

Reply:
I agree, will move to another function